engine/executor/dm/task.go

```diff
@@ -0,0 +1,253 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dm
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tiflow/dm/dm/config"
+	dmconfig "github.com/pingcap/tiflow/dm/dm/config"
+	"github.com/pingcap/tiflow/dm/pkg/log"
+	"github.com/pingcap/tiflow/engine/executor/dm/unit"
+	"github.com/pingcap/tiflow/engine/jobmaster/dm"
+	"github.com/pingcap/tiflow/engine/jobmaster/dm/metadata"
+	"github.com/pingcap/tiflow/engine/jobmaster/dm/runtime"
+	"github.com/pingcap/tiflow/engine/lib"
+	libModel "github.com/pingcap/tiflow/engine/lib/model"
+	"github.com/pingcap/tiflow/engine/lib/registry"
+	"github.com/pingcap/tiflow/engine/model"
+	dcontext "github.com/pingcap/tiflow/engine/pkg/context"
+	dmpkg "github.com/pingcap/tiflow/engine/pkg/dm"
+	"github.com/pingcap/tiflow/engine/pkg/externalresource/broker"
+	"github.com/pingcap/tiflow/engine/pkg/p2p"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// RegisterWorker is used to register dm task to global registry
+func RegisterWorker() {
+	registry.GlobalWorkerRegistry().MustRegisterWorkerType(lib.WorkerDMDump, newTaskFactory(lib.WorkerDMDump))
+	registry.GlobalWorkerRegistry().MustRegisterWorkerType(lib.WorkerDMLoad, newTaskFactory(lib.WorkerDMLoad))
+	registry.GlobalWorkerRegistry().MustRegisterWorkerType(lib.WorkerDMSync, newTaskFactory(lib.WorkerDMSync))
+}
+
+// taskFactory create dm task
+type taskFactory struct {
+	workerType libModel.WorkerType
+}
+
+// newTaskFactory creates abstractFactory
+func newTaskFactory(workerType libModel.WorkerType) *taskFactory {
+	return &taskFactory{workerType: workerType}
+}
+
+// DeserializeConfig implements WorkerFactory.DeserializeConfig
+func (f taskFactory) DeserializeConfig(configBytes []byte) (registry.WorkerConfig, error) {
+	cfg := &dmconfig.SubTaskConfig{}
+	err := cfg.Decode(string(configBytes), true)
+	return cfg, err
+}
+
+// NewWorkerImpl implements WorkerFactory.NewWorkerImpl
+func (f taskFactory) NewWorkerImpl(ctx *dcontext.Context, workerID libModel.WorkerID, masterID libModel.MasterID, conf lib.WorkerConfig) (lib.WorkerImpl, error) {
+	baseDMTask := newBaseTask(ctx, masterID, f.workerType, conf)
+	switch f.workerType {
+	case lib.WorkerDMDump:
+		return newDumpTask(baseDMTask), nil
+	case lib.WorkerDMLoad:
+		return newLoadTask(baseDMTask), nil
+	case lib.WorkerDMSync:
+		return newSyncTask(baseDMTask), nil
+	default:
+		return nil, errors.Errorf("unexpected worker type %d", f.workerType)
+	}
+}
+
+// task defines the interface for dump/load/sync
+type task interface {
+	onInit(ctx context.Context) error
+	onFinished(ctx context.Context) error
+	createUnitHolder(cfg *config.SubTaskConfig) unit.Holder
+}
+
+// baseTask implements some default methods for dm task
+type baseTask struct {
```

Comment:
In my view this struct should named as baseDMWorker. Currently worker and task is 1:1, and they are semanticallly similar like they are both "runnable", can "init", but it's more reasonable that only worker can "send/receive message", and names like `MustRegisterWorkerType`, `lib.WorkerImpl` prefer worker not task.

So worker holding task is easier to understand. Maybe worker holding changefeed, worker holding actors are also acceptable for CDC? @liuzix